src/js_sync.hpp

```diff
@@ -762,6 +765,18 @@ void SubscriptionClass<T>::get_error(ContextType ctx, ObjectType object, ReturnV
     }
 }
 
+template<typename T>
+void SubscriptionClass<T>::get_name(ContextType ctx, ObjectType object, ReturnValue &return_value) {
+    auto subscription = get_internal<T, SubscriptionClass<T>>(object);
+    if (subscription->m_name == util::none) {
+        // FIXME: should we reconstruct the name to match the one stored in __ResultSets?
```

Comment:
I don't think there's currently a way to get an existing subscription, so not sure if we need to do anything there.

Reply:
My concern is that for unnamed subscriptions `subscription.name` will be `undefined` while `Realm.subscriptions()` will return a list of objects where `name` is not `undefined` but a name constructed by object store. You can see the inconsistency on the test 😞 .